core/src/xtdb/kv/index_store.clj

```diff
@@ -170,9 +170,9 @@
   (^org.agrona.MutableDirectBuffer [b entity content-hash attr]
    (encode-ecav-key-to b entity content-hash attr mem/empty-buffer))
   (^org.agrona.MutableDirectBuffer [^MutableDirectBuffer b ^DirectBuffer entity ^DirectBuffer content-hash ^DirectBuffer attr ^DirectBuffer v]
-   (assert (or (zero? (.capacity attr)) (= c/id-size (.capacity attr)))
+   (assert (or (zero? (.capacity attr)) (<= c/id-size (.capacity attr)))
```

Comment:
OOI - why would these be anything apart from zero or `c/id-size`?

Reply:
Good spot @jarohen. Since https://github.com/xtdb/xtdb/commit/898e7c992b57fe4a4b69713900c7945e5a7b4483 we can revert this assert to what it was doing - I have retested and made the revert.